File: 20-29/problem20.pl
use lib "../Utils";
use bigint;
require Math;

# Main

$res =find_sum_of_digits();
print_result($res);

sub find_sum_of_digits {
  $factorial = Math::factorial(100);
  $sum = Math::sum_of_digits($factorial);
  return $sum;
}

sub print_result {
  $res = shift;
  print "The sum of the digits in the number 100! is $res.\n";
}

File: 20-29/problem20.fs
let factorial (n:bigint) = [bigint(1)..n] |> List.reduce (*)
let sumOfDigits n =
    string(n).ToCharArray() 
    |> Array.map (fun c -> int c - int '0')
    |> Array.sum
    

let number = bigint 100
printfn "%A" (sumOfDigits (factorial number))
